werkzeug/serving.py

```diff
@@ -250,7 +250,7 @@ def version_string(self):
         return BaseHTTPRequestHandler.version_string(self).strip()
 
     def address_string(self):
-        return self.client_address[0]
+        return self.environ['REMOTE_ADDR']
```

Comment:
Not sure how defensive you'd care to be here, since technically it'd be possible to raise an `AttributeError` if `self.environ` hasn't been set yet. I can't think of a scenario where `self.address_string` is called without calling `run_wsgi` first.

Or is there a concern with a mutation that deletes the `REMOTE_ADDR` key? I assume if somebody did that, that'd be a whole cascade of issues.